src/multiplicity/mod.rs

```diff
@@ -0,0 +1,824 @@
+//! # Given a Relation representing a computaion on a sampled table, a table (representing the schema of original dataset) and a weight representing
+//!
+//! This is experimental and little tested yet.
+//!
+use crate::{
+    builder::{Ready, With},
+    expr::{aggregate, identifier::Identifier, Expr},
+    hierarchy::Hierarchy,
+    relation::{Join, Map, Reduce, Relation, Set, Table, Variant as _, Visitor},
+    visitor::Acceptor,
+    WithIterator,
+};
+use std::{error, fmt, rc::Rc, result, vec};
+
+#[derive(Debug, Clone)]
+pub enum Error {
+    Other(String),
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Error::Other(err) => writeln!(f, "{}", err),
+        }
+    }
+}
+
+impl error::Error for Error {}
+
+pub type Result<T> = result::Result<T, Error>;
+
+#[derive(Clone, Debug)]
+pub struct RelationWithMultiplicity(Relation, f64);
+
+/// A visitor to compute RelationWithMultiplicity propagationing table weights
+/// and LINE_WEIGHT.
```

Comment:
**Line** is ambiguous (as it may refer to 1D vector spaces) use **Rows** and **Columns**.

Reply:
changed with LINE_WEIGHT with ROW_WEIGHT